mllib/src/main/scala/org/apache/spark/mllib/recommendation/MatrixFactorizationModel.scala

```diff
@@ -165,7 +183,7 @@ class MatrixFactorizationModel @Since("0.8.0") (
    */
   @Since("1.1.0")
   def recommendProducts(user: Int, num: Int): Array[Rating] =
```

Comment:
I'm new to the code base, I understand `Either` and `Option` isn't used a lot in public APIs in Spark but shouldn't it be annotated that the functions throws a certain type of exception (being more explicit with the Exception) ?

like `@throws(classOf[IllegalArgumentException])` ?

Reply:
Returning `Option` could be reasonable, but, I think the caller would end up giving up with an exception anyway. There isn't an obviously better thing to do.

Exceptions all behave as if unchecked in Scala and don't get expressed in method signatures in the byte code. `@throws` is really for Java compatibility where it's important for the Java language to express that a checked exception is thrown (it makes a compile-time difference in Java).

That's not necessary here, but scaladoc'ing exceptions like this is indeed a good idea.

Reply:
Thanks for the explanation Sean :)